### PR TITLE
GMime

### PR DESCRIPTION
**Pull request by vhdirk**
This accompagnies my PR at https://github.com/gtk-rs/sys/pull/87
I'm not sure the line in `fix.sh` is what ultimately ended up 'fixing' the gir file. xmlstarlet is a weird thing.

### CODE REVIEW

File: fix.sh [hunk 1]
```diff
@@ -35,3 +35,8 @@ xmlstarlet ed -P \
 	-i '//_:field[@name="phase"]/_:type[@name="TouchpadGesturePhase"]' -t attr -n 'c:type' -v 'gint8' \
 	Gdk-3.0.gir > tmp
 mv tmp Gdk-3.0.gir
+
+xmlstarlet ed -P \
+	-i '//*[@type="iconv_t"]' -t attr -n 'name' -v 'iconv_t' \
```
Comment (EPashkin) [inline fix.sh]: Not `@c:type` ?

Comment (EPashkin) [reply fix.sh]: And current command don't do anything :( Even after changing c:type

Comment (EPashkin) [reply fix.sh]: `-i '//_:type[@c:type="iconv_t"]' -t attr -n 'name' -v 'gpointer'` produce almost same GMime-3.0.gir for me

Comment (vhdirk) [reply fix.sh]: Thanks.
With using -L and @c:type, it now does work as expected.

Comment (vhdirk) [reply fix.sh]: I'd assume that using gpointer here breaks any change of using a real iconv binding later on?

File: fix.sh [hunk 2]
```diff
@@ -35,3 +35,7 @@ xmlstarlet ed -P \
 	-i '//_:field[@name="phase"]/_:type[@name="TouchpadGesturePhase"]' -t attr -n 'c:type' -v 'gint8' \
 	Gdk-3.0.gir > tmp
 mv tmp Gdk-3.0.gir
+
+xmlstarlet ed -P -L \
+	-i '//*[@c:type="iconv_t"]' -t attr -n 'name' -v 'iconv_t' \
```
Comment (EPashkin) [inline fix.sh]: Currently it add name on any run :(
`type[@c:type="iconv_t" and not(@name)]` seems fix this too

**Comment by EPashkin** [comment]
Only problem that now library "incomplete" from gir point of view
`thread 'main' panicked at 'Incomplete library, unresolved: ["GMime.iconv_t"]', src\library_postprocessing.rs:55:13`

**Comment by EPashkin** [comment]
About replacing 'iconv_t' with `gpointer`:
sys use only `c:type` so adding manual `iconv_t` as you do or reexport from real iconv will work (by way go to manual.rs),
but there will be problems in normal mode.
Maybe you need add new type `iconv_t` here as alias for `gpointer`.

**Comment by vhdirk** [comment]
I switched it to a gpointer now, since I don't know how to add a type alias in a gir file.
I'll revisit this later when I get everything else working in gmime-rs :)

**Comment by tmiasko** [comment]
LGTM

Using gpointer (which is just void*) should work fine given that what iconv_t
type actually is. rust-iconv uses that as well. Alias just gives it different,
a little bit nicer name, for example:

```
pub type iconv_t = *mut void;
```

**Comment by vhdirk** [comment]
Closing this since I renamed the source branch to 'gmime'

**Pull request closed without merging**